src/main/java/gift/repository/ProductRepository.java

```diff
@@ -5,57 +5,55 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.NoSuchElementException;
 import java.util.Optional;
 import org.springframework.stereotype.Repository;
 
 
 @Repository
 public class ProductRepository {
 
-  private final Map<Long, Product> products = new HashMap<>();
-  private Long sequence = 2L; // ID 자동 증가용 (초기값 2, 이미 1번 있음)
+    private final Map<Long, Product> products = new HashMap<>();
+    private Long sequence = 1L;
 
-  public ProductRepository() {
-    // 초기 더미 데이터
-    products.put(1L, new Product(1L, "아이스 카페 아메리카노 T", 4500,
-        "https://st.kakaocdn.net/product/gift/product/20231010111814_9a667f9eccc943648797925498bdd8a3.jpg"));
-  }
-
-  public List<Product> findAll() {
-    return new ArrayList<>(products.values());
-  }
+    public ProductRepository() {
+    }
 
-  public Optional<Product> findById(Long id) {
-    return Optional.ofNullable(products.get(id));
-  }
+    public List<Product> findAll() {
+        return new ArrayList<>(products.values());
+    }
 
+    public Optional<Product> findById(Long id) {
+        return Optional.ofNullable(products.get(id));
+    }
 
-  public Product save(Product product) {
-    Long id = sequence++;
-    Product newProduct = new Product(id, product.getName(), product.getPrice(),
-        product.getImageUrl());
-    products.put(id, newProduct);
-    return newProduct;
-  }
 
-  public Product update(Long id, String name, int price, String imageUrl) {
-    Product product = products.get(id);
-    if (product == null) {
-      return null; // 없는 ID는 무시
+    public Product save(Product product) {
+        Long id = sequence++;
+        Product newProduct = new Product(product.getName(), product.getPrice(),
+                product.getImageUrl());
+        newProduct.setId(id);
+        products.put(id, newProduct);
+        return newProduct;
     }
 
-    Product updated = new Product(id, name, price, imageUrl);
-    products.put(id, updated);
-    return updated;
-  }
+    public Product update(Long id, String name, int price, String imageUrl) {
+        Product product = products.get(id);
+        if (product == null) {
+            throw new NoSuchElementException("해당 ID의 상품이 존재하지 않습니다...");
+        }
 
-  public boolean delete(Long id) {
-    if (!products.containsKey(id)) {
-      return false;
+        Product updated = new Product(name, price, imageUrl);
+        updated.setId(id);
```

Comment:
업데이트를 새로운 객체를 만들어서 대체하실꺼라면 
주생성자를 통해서 id까지 할당을 받는게 조금 더 깔끔하지 않을까요? 😃


```suggestion
        Product updated = new Product(id, name, price, imageUrl);
```

Reply:
지당하신 말씀입니다. 이전 코드는 일을 두번 나눠했네요..!